modules/rgbd/perf/perf_tsdf.cpp

```diff
@@ -219,4 +219,54 @@ PERF_TEST(Perf_TSDF, raycast)
     SANITY_CHECK_NOTHING();
 }
 
+PERF_TEST(Perf_HashTSDF, integrate)
+{
+    Ptr<kinfu::Params> _params;
+    _params = kinfu::Params::hashTSDFParams(true);
+
+    Ptr<kinfu::Volume> volume = kinfu::makeVolume(_params->volumeType, _params->voxelSize, _params->volumePose.matrix,
+        _params->raycast_step_factor, _params->tsdf_trunc_dist, _params->tsdf_max_weight,
+        _params->truncateThreshold, _params->volumeDims);
+
+    Ptr<Scene> scene = Scene::create(_params->frameSize, _params->intr, _params->depthFactor);
+    std::vector<Affine3f> poses = scene->getPoses();
+
+    for (size_t i = 0; i < poses.size(); i++)
+    {
+        UMat _points, _normals;
+        Matx44f pose = poses[i].matrix;
+        Mat depth = scene->depth(pose);
+        startTimer();
+        volume->integrate(depth, _params->depthFactor, pose, _params->intr);
+        stopTimer();
+    }
+    SANITY_CHECK_NOTHING();
+}
+
+PERF_TEST(Perf_HashTSDF, raycast)
+{
+    Ptr<kinfu::Params> _params;
+    _params = kinfu::Params::hashTSDFParams(true);
+
+    Ptr<kinfu::Volume> volume = kinfu::makeVolume(_params->volumeType, _params->voxelSize, _params->volumePose.matrix,
+        _params->raycast_step_factor, _params->tsdf_trunc_dist, _params->tsdf_max_weight,
+        _params->truncateThreshold, _params->volumeDims);
+
+    Ptr<Scene> scene = Scene::create(_params->frameSize, _params->intr, _params->depthFactor);
+    std::vector<Affine3f> poses = scene->getPoses();
+
+    for (size_t i = 0; i < poses.size(); i++)
+    {
+        UMat _points, _normals;
+        Matx44f pose = poses[i].matrix;
+        Mat depth = scene->depth(pose);
+
+        volume->integrate(depth, _params->depthFactor, pose, _params->intr);
+        startTimer();
+        volume->raycast(pose, _params->intr, _params->frameSize, _points, _normals);
+        stopTimer();
+    }
+    SANITY_CHECK_NOTHING();
+}
```

Comment:
Can these 2 functions be joined together as it was done in accuracy tests?

Reply:
no, i tried :(

Reply:
done, but partially